### PR TITLE
Rtd

### PR DESCRIPTION
**Pull request by jburel**
Build the doc for readthedocs see https://jburel-omeroweb-install.readthedocs.io/en/latest/. History has been preserved.

* This PR does not aim at changing the build system. 
* The walkthroughs and the nginx.conf files are generated on the fly i.e. no longer under version control.
*  This replaces some of the logic of https://github.com/ome/omero-documentation/blob/develop/omero/autogen_docs. This means that building the doc has ``omero-web`` and ``ansible`` as dependencies.
* I have used latest on my fork but stable should be used.

 webconfig parameters page could be added

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+------------
+
+**Recommended:**
+
+
+:doc:`walkthrough/omeroweb-install-centos7-ice3.6`
+  Instructions for installing OMERO.web from scratch on CentOS 7 with Ice 3.6.
+
+:doc:`walkthrough/omeroweb-install-debian10-ice3.6`
+  Instructions for installing OMERO.web from scratch on Debian 10 with Ice 3.6.
```
Comment (dominikl) [inline docs/index.rst:50]: A bit strange that we list Debian 10 under "Recommended" and Debian 9 under "Upcoming". Shouldn't it be the other way round?

Comment (jburel) [reply docs/index.rst:50]: This is a ``copy & paste`` from ome-doc, we can certainly fix it.
The approach proposed here will simplify the release of such change at no cost.

**Comment by dominikl** [comment]
Looks good, rtd docs match our existing docs 👍 https://github.com/ome/omeroweb-install/pull/75#discussion_r607662868 could be changed before moving to rtd, but doeesn't have to (isn't really the subject of this PR).

**Comment by joshmoore** [comment]
@jburel : what do you mean "History has been preserved."? What steps did you take to get here?

**Comment by jburel** [comment]
I used ``git filter-branch --subdirectory-filter <dir> -- --all`` in ome-documentation to filter the files to move to this repo.
This is what I meant

**Comment by joshmoore** [comment]
Ok. Looking at https://github.com/ome/omero-documentation/pull/2180 I guess I understand: web content from ome(ro)-documentation is now here with all of its history.

So this repo would be renamed "omero-web-documentation"?

**Comment by sbesson** [comment]
> So this repo would be renamed "omero-web-documentation"?

I guess that's my question as well. In addition to the walkthroughs which sources were already in this repository, this PR pulls some pages from the system administrator documentation (like the customization and public) but not others (like upgrade, the configuration properties etc). Is the intent to extract everything that's specific to OMERO.web out of https://docs.openmicroscopy.org/omero/5.6.3/sysadmins/index.html ?

**Comment by jburel** [comment]
* Repo should be renamed to reflect closely its content similar to what happens to ``ome-doc`` after removing ``contributing``
* My intent was to migrate web-installation related documentation currently under ``sysadmin`` to that repository. I did not plan to migrate the limited installation info under ``developers``
* For configuration, there is the overarching list in ``omero-doc`` but web specific could be here too. same for upgrade. Up for debate

**Comment by jburel** [comment]
Another option could be to keep ``omero-documentation`` and have some of the strategy used in this repo in ``conf.py``
i.e.
* clone this repo and omero-install
* generate the files on the fly

This will means that building ``omero-documentation`` will require several dependencies namely sphinx, omero-web, ansible.

**Comment by jburel** [comment]
3rd option:
* We generate and publish the walkthrough in readthedocs from that repository. Basically we will have an index page and conf.py and keep the rest in omero-documentation

**Comment by sbesson** [comment]
The three options are useful to help settling the decision, thanks @jburel. Trying to summarize my questions, I see two main question that need to be answered and will define how it should be named

- which content  should live in this documentation (i.e. how should we name it). Options are:
  - installation
  - configuration
  - usage e.g. the JSON API
  - development e.g. building Web apps
- which type of  documentation should be included as defined in https://documentation.divio.com/. I would consider the walkthrough examples as how-to guides while the configuration properties page is more reference. Do we want to mix the two types of documentation under a single umbrella or maintain them separately?

**Comment by joshmoore** [comment]
Few questions:
 * What was decided on the naming front?
 * Do any build dependencies remain between the different repos?

**Comment by jburel** [comment]
For build dependencies: we need to sphinx (usual), omero-web (to generate config) and ansible (current status). Ansible could be dropped but not as part of this effort.

We did not discuss the name of the repo. This repo will host the installation/config of omero-web not the usage of API so we should probably have "installation" in the name

**Comment by joshmoore** [comment]
> For build dependencies:

Guess I meant in which order must the different repositories be built.

**Comment by jburel** [comment]
In that case. This repository only relies on published tag i.e. 
* We push a new omero-web version for example that changes the configuration
* We will need to run the doc build so the new configuration parameters are exposed in the doc

**Comment by jburel** [comment]
Alternative to my previous comment:
the file describing web configuration property is in omero-web repo as reference documentation
and we have a link to such page.
This will prevent to have to rebuild the doc hosted in this repo if a new property is added.

**Comment by jburel** [comment]
I am in favour of **not** including config.rst in this repository, this will create a dependance of omero-web.
Such config file should be in the "reference" doc for  omero-web (to be created) 
In the meantime, I think I will point to https://docs.openmicroscopy.org/omero/5.6.3/sysadmins/config.html#web

**Comment by jburel** [comment]
With the option taken, there should not be any dependency i.e. build after releasing omero-web

**Pull request closed without merging**